src/transformer_deploy/utils/t5_utils.py

```diff
@@ -0,0 +1,826 @@
+        output_path=decoder_if_fp16_model_path,
+    )
+
+    torch.cuda.empty_cache()
+    gc.collect()
+
+    # Check ONNX decoder output: Zero copy output
+    """
+    Below, we check that the new model output is similar to the ones from Pytorch.
+    We use our new implementation of inference call.
+    The idea is the following:
+        - we ask ONNX Runtime to output a pointer to the CUDA array containing the result of the inference;
+        - we use Cupy API to wrap the array and provide information regarding tensor shape and type.
+         Cupy doesn't own the data;
+        - we use Dlpack support to convert the Cupy tensor to Pytorch, another zero copy process.
+    This pipeline is unsafe, as the content of the tensor may change or disappear silently: only ONNX Runtime has the
+    control of the array containing the data. It will happen at the next inference call. Because we know that during
+    the text generation we discard each output before recalling ONNX Runtime, it works well in our case.
+    A second benefit of this approach is that we do not have anymore to guess the output shape.
+    """
+    model_pytorch = model_pytorch.cuda()
+    model_decoder = model_decoder.cuda()
+    input_ids = input_ids.cuda()
+    model_pytorch = model_pytorch.eval()
+    model_decoder = model_decoder.eval()
+    with torch.inference_mode():
+        out_encoder_pytorch: BaseModelOutputWithPastAndCrossAttentions = model_pytorch.encoder(input_ids=input_ids)
+        previous_step_pytorch: BaseModelOutputWithPastAndCrossAttentions = model_decoder(
+            input_ids=input_ids[:, :-1], encoder_hidden_states=out_encoder_pytorch.last_hidden_state
+        )
+        out_decoder_pytorch: BaseModelOutputWithPastAndCrossAttentions = model_decoder(
+            input_ids=input_ids, encoder_hidden_states=out_encoder_pytorch.last_hidden_state
+        )
+    model_pytorch = model_pytorch.cpu()
+    torch.cuda.empty_cache()
+    decoder_onnx = create_model_for_provider(decoder_if_fp16_model_path, "CUDAExecutionProvider", log_severity=3)
+    out_decoder_onnx_no_cache = decoder_onnx_inference(
+        decoder_input_ids=input_ids,
+        encoder_hidden_states=out_encoder_pytorch.last_hidden_state.half(),
+        enable_cache=torch.tensor([False], device="cuda", dtype=torch.bool),
+        past_key_values=None,
+        decoder_onnx=decoder_onnx,
+        num_layers=model_pytorch.config.num_layers,
+    )
+    are_equal(
+        a=out_decoder_onnx_no_cache.last_hidden_state[:, -1:, :],
+        b=out_decoder_pytorch.last_hidden_state[:, -1:, :],
+    )
+    # check that past states are identical between ONNX and Pytorch
+    assert len(out_decoder_onnx_no_cache.past_key_values) == len(out_decoder_pytorch.past_key_values)
+    for (o_dec_k, o_dev_v, o_enc_k, o_enc_v), (p_dec_k, p_dev_v, p_enc_k, p_enc_v) in zip(
+        out_decoder_onnx_no_cache.past_key_values, out_decoder_pytorch.past_key_values
+    ):
+        are_equal(a=o_dec_k, b=p_dec_k)
```

Comment:
Do we need to do all quality check ? The test in convert isn't enough ?
In the other convert_to_onnx we don't do this. So if we keed something like that it should be extracted IMO

Reply:
I removed the quality check (I think the test in conversion is enough)